python/pyspark/sql/tests.py

```diff
@@ -3383,6 +3400,43 @@ def test_vectorized_udf_varargs(self):
         res = df.select(f(col('id')))
         self.assertEquals(df.collect(), res.collect())
 
+    def test_vectorized_udf_timestamps(self):
+        from pyspark.sql.functions import pandas_udf, col
+        from datetime import date, datetime
+        schema = StructType([
+            StructField("idx", LongType(), True),
+            StructField("date", DateType(), True),
+            StructField("timestamp", TimestampType(), True)])
+        # TODO Fails with time before epoch: (0, date(1969, 1, 1), datetime(1969, 1, 1, 1, 1, 1))
+        data = [(0, date(1985, 1, 1), datetime(1985, 1, 1, 1, 1, 1)),
+                (1, date(2012, 2, 2), datetime(2012, 2, 2, 2, 2, 2)),
+                (2, date(2100, 3, 3), datetime(2100, 3, 3, 3, 3, 3)),
+                (3, date(2104, 4, 4), datetime(2104, 4, 4, 4, 4, 4))]
+
+        df = self.spark.createDataFrame(data, schema=schema)
+
+        # Check that a timestamp passed through a pandas_udf will not be altered by timezone calc
+        identity = pandas_udf(lambda t: t, returnType=TimestampType())
+        df = df.withColumn("timestamp_copy", identity(col("timestamp")))
+
+        @pandas_udf(returnType=BooleanType())
+        def check_data(idx, date, timestamp, timestamp_copy):
+            is_equal = timestamp == timestamp_copy
+            if is_equal.all():
+                for i in xrange(len(is_equal)):
+                    # TODO Fails with tz offset: date[i].date() == data[idx[i]][1] and
```

Comment:
I think the date types are also getting a timezone conversion which cause the equals test here to fail, need to look into this more